pkg/virt-controller/watch/vmi_test.go

```diff
@@ -1605,6 +1607,34 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 
 			controller.Execute()
 		})
+		Context("And the virt-launcher pod has multus annotations", func() {
+			It("should copy a corresponding Pod on VirtualMachineInstance creation", func() {
+				By("creating a vmi in Scheduling phase")
+				vmi := NewPendingVirtualMachine("testvmi")
+				setReadyCondition(vmi, k8sv1.ConditionFalse, virtv1.PodConditionMissingReason)
+				vmi.Status.Phase = virtv1.Scheduling
+
+				By("attaching a virt-launcher pod with multus annotations to the vmi")
+				networkStatusAnnotation := `some-network-status`
+				networkAnnotation := `some-network`
+				pod := NewPodForVirtualMachine(vmi, k8sv1.PodRunning)
+				pod.Annotations[networkv1.NetworkStatusAnnot] = networkStatusAnnotation
+				pod.Annotations[networkv1.NetworkAttachmentAnnot] = networkAnnotation
+
+				addVirtualMachine(vmi)
+				podFeeder.Add(pod)
+				addActivePods(vmi, pod.UID, "")
+
+				vmiInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
+					Expect(arg.(*virtv1.VirtualMachineInstance).Annotations).To(HaveKey(networkv1.NetworkStatusAnnot))
+					Expect(arg.(*virtv1.VirtualMachineInstance).Annotations[networkv1.NetworkStatusAnnot]).To(Equal(networkStatusAnnotation))
+					Expect(arg.(*virtv1.VirtualMachineInstance).Annotations).To(HaveKey(networkv1.NetworkAttachmentAnnot))
+					Expect(arg.(*virtv1.VirtualMachineInstance).Annotations[networkv1.NetworkAttachmentAnnot]).To(Equal(networkAnnotation))
```

Comment:
Ohh my... this is an assertion embedded in the mock expecter.
How about do this:

```golang
vmiInterface.EXPECT().Update(gomock.Any()).Do(func(arg interface{}) {
    vmi = arg.(*virtv1.VirtualMachineInstance)
})
Expect(vmi.Annotations).To(HaveKeyWithValue(networkv1.NetworkStatusAnnot, networkStatusAnnotation))
Expect(vmi.Annotations).To(HaveKeyWithValue(networkv1.NetworkAttachmentAnnot, networkAnnotation))
```

Reply:
Like!